citi-onboarding/DADe-onepage
# Feature/create banner and navbar

## What I did

- Create banner view, it's necessary components and files (.css and .jsx)
- Create navbar view and it's necessary components
- Make the necessary exports and imports to use both of them in the App.jsx file
- Call both of them in the App.jsx file

- Obs: I did them both in a single branch because I needed to use the same button component for the navbar and the banner.  

| Mockup | Real |
|--------|------|
![Captura de Tela_select-area_20210615181405](https://user-images.githubusercontent.com/61971582/122124332-82d3a980-ce05-11eb-80bd-4a3519b08dcc.png) |
![Captura de Tela_select-area_20210615180423](https://user-images.githubusercontent.com/61971582/122124340-87985d80-ce05-11eb-9351-9d9d859c2578.png)

## How to test

1. Open the terminal and go to the folder of this project

2. In terminal:
   1. `git checkout develop`
   2. `git pull`
   3. `git checkout  feature/create-banner-and-navbar`
   4. `git pull origin feature/create-banner-and-navbar`
   5. `npm install`
   6. `cd client/`
   7. `npm start`

3. In your browser go to http://localhost:3000 (should open automatically when running `npm start`)

4. Check if the component looks just like the mockup in different screen sizes (minimum: 1024px width, mobile version will need a media query). Also, check if after expading screen width past 1440px, the navbar items stay in place, instead of the distance between them increasing.

5. In VScode open the `client/src/components/BannerTitleAndButton/BannerTitleAndButton.jsx` file

6. In line 10 of the mentioned file, change the `link="https://google.com"` property to any link, save the file, and check if the Banner button is directing you to the recent modified link after you click it.

7. In the same file mentioned on the topic 5,  in line 9, change the `purple='Faça parte do diretório acadêmico' yellow='de Design da UFPE'`  properties values for any other strings, save the file, and check if the Banner Title is now displaying your new message.

7. Open the file of the following path: `client/src/Views/Banner/Banner.jsx`, and change the line 7 (`import logo from '../../components/LogoDADe/Logo_DADe.svg'`), adding a link to any image from the web as the following `import logo from <"your_image_link">`. Now save the file, and check if the Banner image now displays that same, newly added, image.

8. Read the changed files section on GitHub 

9. Test the things that you think are worth testing, even the ones that are not in this description 

10. Don't commit or push your changes from the tests, as the original informations were the correct ones for the current version of the project. 

&nbsp;
***
**Resolves:** Navbar and Banner issues
